test/unified-config.test.js

```diff
@@ -0,0 +1,453 @@
+
+    doChecks(cfg, {global: config});
+  })
+
+  it('should warn about deprecated config file keys', function () {
+    const config = {
+      traceMode: 'always',
+      sampleRate: 1000000,
+    }
+    writeConfigJSON(config);
+
+    const cfg = guc();
+
+    // fix up to match translated value.
+    config.traceMode = 1;
+    const warnings = [
+      'traceMode is deprecated; it will be invalid in the future',
+      'sampleRate is deprecated; it will be invalid in the future',
+    ];
+    doChecks(cfg, {global: config, warnings});
+  })
+
+  it('should warn about unknown config keys', function () {
+    const config = {badKeyRising: 'i have been a bad key'};
+    writeConfigJSON(config);
+
+    const cfg = guc();
+
+    const unusedConfig = ['badKeyRising: i have been a bad key'];
+    doChecks(cfg, {unusedConfig});
+  })
+
+  it('should not set keys with invalid values', function () {
+    const config = {ec2MetadataTimeout: 'hello', createTraceIdsToken: 'bruce'};
+    writeConfigJSON(config);
+
+    const cfg = guc();
+
+    const warnings = [
+      'invalid configuration file value createTraceIdsToken: bruce',
+      'invalid configuration file value ec2MetadataTimeout: hello',
+    ];
+    doChecks(cfg, {warnings});
+  })
+
+  it('should handle a non-default config file correctly', function () {
+    const config = {ec2MetadataTimeout: 4000};
+    const filename = 'non-default.json';
+    fs.writeFileSync(filename, JSON.stringify(config));
+
+    process.env.APPOPTICS_APM_CONFIG_NODE = filename;
+
+    const cfg = guc();
+
+    fs.unlinkSync(filename);
+
+    doChecks(cfg, {file: `${process.cwd()}/${filename}`, global: config});
+  })
+
+  //
+  // config file errors
+  //
+  it('should correctly report an error reading a non-default config file', function () {
+    const file = process.env.APPOPTICS_APM_CONFIG_NODE = 'xyzzy-bruce.json';
+    const fullpath = `${process.cwd()}/${file}`;
+
+    const cfg = guc();
+
+    const message = 'Cannot find module \'/home/bruce/solarwinds/ao-agent/xyzzy-bruce.json\'';
+    const errors = [
+      `Cannot read config file ${fullpath}: ${message}`
+    ]
+    doChecks(cfg, {file: fullpath, errors});
+  })
+
+  it('should correctly report an error reading a bad config file', function () {
+    writeConfigLiteral('{"i am: bad\n');
+
+    const cfg = guc();
+
+    const errors = [[
+      'Cannot read config file /home/bruce/solarwinds/ao-agent/appoptics-apm: ',
+      '/home/bruce/solarwinds/ao-agent/appoptics-apm.json: Unexpected token \n in JSON at position 11'
+    ].join('')];
+    doChecks(cfg, {errors});
+  })
+
+  //
+  // env vars
+  //
+  it('should warn about unknown APPOPTICS_ environment variables', function () {
+    process.env.APPOPTICS_YINYIN_ROCKS = 'truth';
+
+    const cfg = guc();
+
+    const unusedEnvVars = ['APPOPTICS_YINYIN_ROCKS=truth'];
+    doChecks(cfg, {unusedEnvVars});
+  })
+
+  it('should not set use environment variable values', function () {
```

Comment:
'set use' ? It should not 'set invalid' env var values .... ?

Reply:
thanks.